src/background/stores/assessment-store.ts

```diff
@@ -278,14 +277,12 @@ export class AssessmentStore extends BaseStoreImpl<IAssessmentStoreData> {
     @autobind
     private onUpdateInstanceVisibility(payload: UpdateVisibilityPayload): void {
         const step = this.state.assessmentNavState.selectedTestStep;
-        const config = this.assessmentsProvider.forType(this.state.assessmentNavState.selectedTestType).getVisualizationConfiguration();
-        const assessmentData = config.getAssessmentData(this.state);
-
-        if (assessmentData.generatedAssessmentInstancesMap == null) {
-            return;
-        }
-
         payload.payloadBatch.forEach(updateInstanceVisibilityPayload => {
+            const config = this.assessmentsProvider.forType(updateInstanceVisibilityPayload.test).getVisualizationConfiguration();
+            const assessmentData = config.getAssessmentData(this.state);
+            if (assessmentData.generatedAssessmentInstancesMap == null) {
+                return;
+            }
```

Comment:
This change is not absolutely necessary in this PR but I figured to make it since I was working within the test for this and found it odd that it was not behaving per whatever was sent in the payload. Since this **is hidden behind a non-public feature flag**, I would feel comfortable moving forward with this.